tutorial/sample3/webgl-demo.js

```diff
@@ -1,237 +1,265 @@
-  }
+    void main(void) {
+      gl_FragColor = vColor;
+    }
+  `;
+
+  // Initialize a shader program; this is where all the lighting
+  // for the vertices and so forth is established.
+  const shaderProgram = initShaderProgram(gl, vsSource, fsSource);
+
+  // Collect all the info need to use the shader program.
+  // Look up which attribute our shader program is using
+  // for aVertexPosition and uniform locations.
+  const programInfo = {
+    program: shaderProgram,
+    attribLocations: {
+      vertexPosition: gl.getAttribLocation(shaderProgram, 'aVertexPosition'),
+      vertexColor: gl.getAttribLocation(shaderProgram, 'aVertexColor'),
+    },
+    uniformLocations: {
+      projectionMatrix: gl.getUniformLocation(shaderProgram, 'uProjectionMatrix'),
+      modelViewMatrix: gl.getUniformLocation(shaderProgram, 'uModelViewMatrix'),
+    },
+  };
+
+  // Here's where we call the routine that builds all the
+  // objects we'll be drawing.
+  const buffers = initBuffers(gl);
+
+  // Draw the scene
+  drawScene(gl, programInfo, buffers);
 }
 
 //
 // initBuffers
 //
-// Initialize the buffers we'll need. For this demo, we just have
-// one object -- a simple two-dimensional square.
+// Initialize the buffers we'll need. For this demo, we just
+// have one object -- a simple two-dimensional square.
 //
-function initBuffers() {
+function initBuffers(gl) {
 
-  // Create a buffer for the square's vertices.
+  // Create a buffer for the square's positions.
 
-  squareVerticesBuffer = gl.createBuffer();
+  const positionBuffer = gl.createBuffer();
 
-  // Select the squareVerticesBuffer as the one to apply vertex
+  // Select the positionBuffer as the one to apply buffer
   // operations to from here out.
 
-  gl.bindBuffer(gl.ARRAY_BUFFER, squareVerticesBuffer);
+  gl.bindBuffer(gl.ARRAY_BUFFER, positionBuffer);
 
-  // Now create an array of vertices for the square. Note that the Z
-  // coordinate is always 0 here.
+  // Now create an array of positions for the square.
 
-  var vertices = [
-    1.0,  1.0,  0.0,
-    -1.0, 1.0,  0.0,
-    1.0,  -1.0, 0.0,
-    -1.0, -1.0, 0.0
+  const positions = [
+     1.0,  1.0,
+    -1.0,  1.0,
+     1.0, -1.0,
+    -1.0, -1.0,
   ];
 
-  // Now pass the list of vertices into WebGL to build the shape. We
-  // do this by creating a Float32Array from the JavaScript array,
-  // then use it to fill the current vertex buffer.
+  // Now pass the list of positions into WebGL to build the
+  // shape. We do this by creating a Float32Array from the
+  // JavaScript array, then use it to fill the current buffer.
 
-  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(vertices), gl.STATIC_DRAW);
+  gl.bufferData(gl.ARRAY_BUFFER, new Float32Array(positions), gl.STATIC_DRAW);
 
   // Now set up the colors for the vertices
 
   var colors = [
     1.0,  1.0,  1.0,  1.0,    // white
     1.0,  0.0,  0.0,  1.0,    // red
     0.0,  1.0,  0.0,  1.0,    // green
-    0.0,  0.0,  1.0,  1.0     // blue
```

Comment:
Because I'm a trailing comma person. Why not add the comma? Why should the blue line be different than the other 3 lines?